src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -1446,6 +1455,13 @@ _public_ sd_bus* sd_bus_flush_close_unref(sd_bus *bus) {
         if (!bus)
                 return NULL;
 
+        /* Have to do this before flush() to prevent hang */
+        if (bus->busexec_pid)
```

Comment:
we usually avoid using C's downgrade-to-bool feature for comparing numeric values to 0 unless they actually are boolean-like... Hence please write this with an explicit comparison: `if (bus->busexec_pid > 0)` or so, or even `if (pid_is_valid(bus->busexec_pid) > 0)`